uaclient/cli.py

```diff
@@ -399,10 +408,20 @@ def config_parser(parser):
 def attach_parser(parser):
     """Build or extend an arg parser for attach subcommand."""
     parser.usage = USAGE_TMPL.format(name=NAME, command="attach <token>")
+    parser.formatter_class = argparse.RawDescriptionHelpFormatter
     parser.prog = "attach"
-    parser.description = (
+    base_desc = (
         "Attach this machine to Ubuntu Pro with a token obtained"
-        " from {}".format(defaults.BASE_UA_URL)
+        " from:\n{}".format(defaults.BASE_UA_URL)
+    )
+    parser.description = (
+        base_desc
+        + "\n\n"
+        + (
+            "When running this command without a token, we will use the "
+            "magic-attach\nflow to attach this machine into a Ubuntu Pro "
+            "subsctription without a token\nLearn more at: magic-attach-url"
```

Comment:
I don't think "magic attach" should appear in any user-facing text. Here's a suggestion:
```suggestion
            "When running this command without a token, it will generate a short "
            "code and prompt you to attach the machine to your Ubuntu Pro account "
            "using a web browser."
```